src/vrms_arch/disambiguation.py

```diff
@@ -213,5 +219,10 @@ def __init__(self, db):
                 self.packages.append(Package(pkg.name, AMBIGUOUS_PACKAGES[pkg.name]))
             elif pkg.name.startswith("lib32") and pkg.name[6:] in AMBIGUOUS_PACKAGES:
                 self.packages.append(Package(pkg.name, AMBIGUOUS_PACKAGES[pkg.name[6:]]))
+            elif any([' OR ' in l for l in pkg.licenses]):
+                licenses = []
+                for l in pkg.licenses:
+                    licenses.extend(l.split(' OR '))
+                self.packages.append(Package(pkg.name, licenses))
```

Comment:
I think `OR` should be a special case, we should check if at least one of the `OR` terms is free, and if so, discard the commercial alternatives. At least that's how I'd interpret that?

Funny enough, earlier I was thinking of some over-engineered solution of creating some parser for AND and OR alternative licenses, and resolving them to see if the product can be used with a free license. But honestly just splitting the string on `OR` probably works just as well, no need for a parser.

Reply:
On second thought, doesn't this conflict with things like `GPL2-or-later`?

Reply:
> On second thought, doesn't this conflict with things like `GPL2-or-later`?

Could indeed be if someone declared it as e.g. `custom:GPL2 OR later`, because it is before the cleaning. I agree it’s not the cleanest hack, but for the given packages (five in Manjaro) it works.
```python
cloud-init: ['GPL-3.0-only OR Apache-2.0']
maturin: ['Apache-2.0 OR MIT']
openpgp-card-tools: ['Apache-2.0 OR MIT']
python-maturin: ['Apache-2.0 OR MIT']
snd: ['BSD-2-Clause', 'GPL-2.0-or-later OR GL2PS', 'LLGPL', 'TCL']
```
But I agree that proper treatment would be of course better. This would also include the proper treatment of exceptions and their splitting word `WITH`.

If you prefer I can also remove this addition for now. And in future we can think about how to implement this properly. Let me know!